mlir/lib/IR/PatternMatch.cpp

```diff
@@ -158,6 +158,12 @@ void RewriterBase::eraseOp(Operation *op) {
   assert(op->use_empty() && "expected 'op' to have no uses");
   auto *rewriteListener = dyn_cast_if_present<Listener>(listener);
 
+  // If the current insertion point is before the erased operation, we adjust
+  // the insertion point to be after the operation.
+  if (getInsertionBlock() == op->getBlock() &&
+      getInsertionPoint() == op->getIterator())
```

Comment:
```suggestion
  if (getInsertionPoint() == op->getIterator())
```
nit: Just comparing iterators ought to be enough